modules/layers/src/text-layer/utils.js

```diff
@@ -69,25 +69,119 @@ export function buildMapping({
   };
 }
 
-export function transformRow(row, iconMapping, lineHeight) {
+function getTextWidth(text, mapping) {
+  const textArray = Array.from(text);
+  let width = 0;
+  for (let i = 0; i < textArray.length; i++) {
+    const character = textArray[i];
+    let frameWidth = null;
+    const frame = mapping[character];
+    if (frame) {
+      frameWidth = frame.width;
+    } else {
+      log.warn(`Missing character: ${character}`)();
+      frameWidth = MISSING_CHAR_WIDTH;
+    }
+
+    width += frameWidth;
+  }
+
+  return width;
+}
+
+// eslint-disable-next-line max-statements
+export function autoWrapping(text, wordBreak, maxWidth, iconMapping) {
+  if (!text) {
+    return null;
+  }
+
+  // 1. break text into groups
+  let groups = null;
+  if (wordBreak === 'break-word') {
+    let index = 0;
+    // split words with white space
+    groups = text.split(' ').map(g => {
```

Comment:
You can merge this map operation into the for loop below and avoid creating these small objects.